src/connection_pool.rs

```diff
@@ -1007,7 +1065,10 @@ fn rebalancer_loop<C, F>(
                 protected_data.clone(),
             );
 
-            debug!(log, "Connection rebalance completed");
+            debug!(
+                log,
+                "Connection rebalance completed: {:#?}", rebalance_result
```

Comment:
I don't think it needs to be done for this change, but perhaps we should wrap the left type of the `Result` returned from `rebalance_connections` (`Option<HashMap<BackendKey, ConnectionCount>>`) in a new type and add a `Display` or `Debug` instance.